modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -1140,12 +1193,18 @@ struct InferList: public cv::detail::KernelTag {
             return;
         }
 
-        IE::Blob::Ptr this_blob = extractBlob(*ctx, 1);
+        // NB: This blob will be used to make roi from its, so
+        // it should be treated as image
+        IE::Blob::Ptr this_blob = extractBlob(*ctx, 1, cv::gapi::ie::TraitAs::IMAGE);
 
         std::vector<std::vector<int>> cached_dims(ctx->uu.params.num_out);
         for (auto i : ade::util::iota(ctx->uu.params.num_out)) {
-            const IE::DataPtr& ie_out = ctx->uu.outputs.at(ctx->uu.params.output_names[i]);
-            cached_dims[i] = toCV(ie_out->getTensorDesc().getDims());
+            const auto& out_name = ctx->uu.params.output_names[i];
+            const auto& desc =
+                ctx->uu.params.kind == cv::gapi::ie::detail::ParamDesc::Kind::Load
+                    ? ctx->uu.net.getOutputsInfo().at(out_name)->getTensorDesc()
+                    : ctx->uu.this_network.GetOutputsInfo().at(out_name)->getTensorDesc();
```

Comment:
Maybe we should really locate this behavior in different classes wrapping this behavior for `Load` and `Import` cases, and don't have all these switches scattered in the whole file.

Reply:
But please don't overdesign it if you plan to, maybe we should put it as an input to our overall future IE backend refactoring

Reply:
Yep, my plan was not `overdesign` implementation, I keep it as simple as possible, make sense to refactor it with `OpenVINO 2.0`